### PR TITLE
Add join info to compressed chunk

### LINKED ISSUE #1558: Planner does not enforce join condition on hypertable with compression
**Relevant system information:**
 - OS: Debian 10
 - PostgreSQL version : postgres (PostgreSQL) 11.5 (Debian 11.5-1+deb10u1)
 - TimescaleDB version : 1.5.1
 - Installation method: apt

**Describe the bug**
A JOIN with several conditions between an hypertable with compression and another table selects rows that do not met some of the conditions.

It seems that the planner is wrong on conditions to apply on compressed chunks.

**How I reproduce on my system**

I have a "metrics" table:

```
\d+ metrics
+-----------+--------------------------+----------------------------------+-----------+----------------+---------------+
| Column    | Type                     | Modifiers                        | Storage   | Stats target   | Description   |
|-----------+--------------------------+----------------------------------+-----------+----------------+---------------|
| timestamp | timestamp with time zone |  not null default now()          | plain     | <null>         | <null>        |
| source    | text                     |  not null                        | extended  | <null>         | <null>        |
| metric    | text                     |  not null                        | extended  | <null>         | <null>        |
| value     | double precision         |  default 'NaN'::double precision | plain     | <null>         | <null>        |
+-----------+--------------------------+----------------------------------+-----------+----------------+---------------+
Indexes:
    "metrics_pkey" PRIMARY KEY, btree ("timestamp", source, metric)
    "metrics_source_idx" btree (source, "timestamp" DESC)
    "metrics_timestamp_idx" btree ("timestamp" DESC)
Triggers:
    ts_insert_blocker BEFORE INSERT ON metrics FOR EACH ROW EXECUTE PROCEDURE _timescaledb_internal.insert_blocker()
Child tables: _timescaledb_internal._hyper_1_11_chunk,
              _timescaledb_internal._hyper_1_13_chunk,
              _timescaledb_internal._hyper_1_14_chunk,
              _timescaledb_internal._hyper_1_16_chunk,
              _timescaledb_internal._hyper_1_18_chunk,
              _timescaledb_internal._hyper_1_1_chunk,
              _timescaledb_internal._hyper_1_20_chunk,
              _timescaledb_internal._hyper_1_22_chunk,
              _timescaledb_internal._hyper_1_24_chunk,
              _timescaledb_internal._hyper_1_26_chunk,
              _timescaledb_internal._hyper_1_28_chunk,
              _timescaledb_internal._hyper_1_2_chunk,
              _timescaledb_internal._hyper_1_30_chunk,
              _timescaledb_internal._hyper_1_32_chunk,
              _timescaledb_internal._hyper_1_3_chunk,
              _timescaledb_internal._hyper_1_4_chunk,
              _timescaledb_internal._hyper_1_5_chunk,
              _timescaledb_internal._hyper_1_6_chunk,
              _timescaledb_internal._hyper_1_7_chunk,
              _timescaledb_internal._hyper_1_9_chunk
Has OIDs: no

> select count(*) from metrics;
+-----------+
| count     |
|-----------|
| 247133724 |
+-----------+

```

For which compression is enabled:
```
> select chunk_name, compression_status FROM timescaledb_information.compressed_chunk_stats;
+-----------------------------------------+----------------------+
| chunk_name                              | compression_status   |
|-----------------------------------------+----------------------|
| _timescaledb_internal._hyper_1_3_chunk  | Compressed           |
| _timescaledb_internal._hyper_1_1_chunk  | Compressed           |
| _timescaledb_internal._hyper_1_2_chunk  | Compressed           |
| _timescaledb_internal._hyper_1_4_chunk  | Compressed           |
| _timescaledb_internal._hyper_1_5_chunk  | Compressed           |
| _timescaledb_internal._hyper_1_6_chunk  | Compressed           |
| _timescaledb_internal._hyper_1_20_chunk | Uncompressed         |
| _timescaledb_internal._hyper_1_7_chunk  | Compressed           |
| _timescaledb_internal._hyper_1_22_chunk | Uncompressed         |
| _timescaledb_internal._hyper_1_9_chunk  | Compressed           |
| _timescaledb_internal._hyper_1_24_chunk | Uncompressed         |
| _timescaledb_internal._hyper_1_11_chunk | Compressed           |
| _timescaledb_internal._hyper_1_26_chunk | Uncompressed         |
| _timescaledb_internal._hyper_1_13_chunk | Compressed           |
| _timescaledb_internal._hyper_1_28_chunk | Uncompressed         |
| _timescaledb_internal._hyper_1_14_chunk | Compressed           |
| _timescaledb_internal._hyper_1_30_chunk | Uncompressed         |
| _timescaledb_internal._hyper_1_16_chunk | Compressed           |
| _timescaledb_internal._hyper_1_32_chunk | Uncompressed         |
| _timescaledb_internal._hyper_1_18_chunk | Compressed           |
+-----------------------------------------+----------------------+


```

I also have a toy table:
```
> \d+ nodetime
+------------+--------------------------+-------------+-----------+----------------+---------------+
| Column     | Type                     | Modifiers   | Storage   | Stats target   | Description   |
|------------+--------------------------+-------------+-----------+----------------+---------------|
| job_id     | integer                  |             | plain     | <null>         | <null>        |
| start_time | timestamp with time zone |             | plain     | <null>         | <null>        |
| stop_time  | timestamp with time zone |             | plain     | <null>         | <null>        |
| node       | text                     |             | extended  | <null>         | <null>        |
+------------+--------------------------+-------------+-----------+----------------+---------------+


> select * from nodetime
+----------+------------------------+------------------------+--------+
| job_id   | start_time             | stop_time              | node   |
|----------+------------------------+------------------------+--------|
| 1071205  | 2019-11-28 08:07:07+00 | 2019-11-28 08:49:24+00 | nova-1 |
+----------+------------------------+------------------------+--------+

```

I try to select rows in "metrics" table with respect to "nodetime" content:
```
> select timestamp, start_time, stop_time, node, source, value from nodetime inner join metrics on nodetime.node = metrics.source and metrics.timestamp > nodetime.start_time and metrics.timestamp < nodetime.stop_time limit 5;
+-------------------------------+------------------------+------------------------+--------+----------+---------+
| timestamp                     | start_time             | stop_time              | node   | source   | value   |
|-------------------------------+------------------------+------------------------+--------+----------+---------|
| 2019-11-25 11:37:59.10852+00  | 2019-11-28 08:07:07+00 | 2019-11-28 08:49:24+00 | nova-1 | nova-1   | 0.0     |
| 2019-11-25 11:37:54.042746+00 | 2019-11-28 08:07:07+00 | 2019-11-28 08:49:24+00 | nova-1 | nova-1   | 0.0     |
| 2019-11-25 11:37:49.106658+00 | 2019-11-28 08:07:07+00 | 2019-11-28 08:49:24+00 | nova-1 | nova-1   | 0.0     |
| 2019-11-25 11:37:44.043496+00 | 2019-11-28 08:07:07+00 | 2019-11-28 08:49:24+00 | nova-1 | nova-1   | 0.0     |
| 2019-11-25 11:37:39.012697+00 | 2019-11-28 08:07:07+00 | 2019-11-28 08:49:24+00 | nova-1 | nova-1   | 0.0     |
+-------------------------------+------------------------+------------------------+--------+----------+---------+
```

However, we see that "metrics.timestamp > nodetime.start_time" condition is not met.

Indeed, it seems the planner does not apply this condition on compressed chunk:
```
> explain select timestamp, start_time, stop_time, node, source, value from nodetime inner join metrics on nodetime.node = metrics.source and metrics.timestamp > nodetime.start_time and metrics.timestamp < nodeti me.stop_time limit 5;
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| QUERY PLAN                                                                                                                                                                 |
|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Limit  (cost=0.08..0.66 rows=5 width=81)                                                                                                                                   |
|   ->  Nested Loop  (cost=0.08..16228666.68 rows=138841916 width=81)                                                                                                        |
|         ->  Seq Scan on nodetime  (cost=0.00..20.20 rows=1020 width=48)                                                                                                    |
|         ->  Append  (cost=0.08..7319.86 rows=859058 width=33)                                                                                                              |
|               ->  Custom Scan (DecompressChunk) on _hyper_1_1_chunk  (cost=0.08..0.39 rows=5000 width=32)                                                                  |
|                     ->  Index Scan using compress_hyper_2_10_chunk__compressed_hypertable_2_source__ts_m on compress_hyper_2_10_chunk  (cost=0.28..0.39 rows=5 width=100)  |
|                           Index Cond: (source = nodetime.node)                                                                                                             |
|               ->  Custom Scan (DecompressChunk) on _hyper_1_2_chunk  (cost=0.03..1.09 rows=35000 width=33)                                                                 |
|                     ->  Index Scan using compress_hyper_2_12_chunk__compressed_hypertable_2_source__ts_m on compress_hyper_2_12_chunk  (cost=0.28..1.09 rows=35 width=101) |
|                           Index Cond: (source = nodetime.node)                                                                                                             |
|               ->  Custom Scan (DecompressChunk) on _hyper_1_3_chunk  (cost=0.03..1.81 rows=67000 width=33)                                                                 |
|                     ->  Index Scan using compress_hyper_2_8_chunk__compressed_hypertable_2_source__ts_me on compress_hyper_2_8_chunk  (cost=0.29..1.81 rows=67 width=101)  |
|                           Index Cond: (source = nodetime.node)                                                                                                             |
|               ->  Custom Scan (DecompressChunk) on _hyper_1_4_chunk  (cost=0.03..1.97 rows=77000 width=33)                                                                 |
|                     ->  Index Scan using compress_hyper_2_15_chunk__compressed_hypertable_2_source__ts_m on compress_hyper_2_15_chunk  (cost=0.29..1.97 rows=77 width=101) |
|                           Index Cond: (source = nodetime.node)                                                                                                             |
|               ->  Custom Scan (DecompressChunk) on _hyper_1_5_chunk  (cost=0.03..2.01 rows=79000 width=33)                                                                 |
|                     ->  Index Scan using compress_hyper_2_17_chunk__compressed_hypertable_2_source__ts_m on compress_hyper_2_17_chunk  (cost=0.29..2.01 rows=79 width=101) |
|                           Index Cond: (source = nodetime.node)                                                                                                             |
|               ->  Custom Scan (DecompressChunk) on _hyper_1_6_chunk  (cost=0.03..2.01 rows=79000 width=33)                                                                 |
|                     ->  Index Scan using compress_hyper_2_19_chunk__compressed_hypertable_2_source__ts_m on compress_hyper_2_19_chunk  (cost=0.29..2.01 rows=79 width=101) |
|                           Index Cond: (source = nodetime.node)                                                                                                             |
|               ->  Custom Scan (DecompressChunk) on _hyper_1_7_chunk  (cost=0.03..1.65 rows=60000 width=33)                                                                 |
|                     ->  Index Scan using compress_hyper_2_21_chunk__compressed_hypertable_2_source__ts_m on compress_hyper_2_21_chunk  (cost=0.29..1.65 rows=60 width=101) |
|                           Index Cond: (source = nodetime.node)                                                                                                             |
|               ->  Custom Scan (DecompressChunk) on _hyper_1_9_chunk  (cost=0.03..1.85 rows=68000 width=33)                                                                 |
|                     ->  Index Scan using compress_hyper_2_23_chunk__compressed_hypertable_2_source__ts_m on compress_hyper_2_23_chunk  (cost=0.29..1.85 rows=68 width=101) |
|                           Index Cond: (source = nodetime.node)                                                                                                             |
|               ->  Custom Scan (DecompressChunk) on _hyper_1_11_chunk  (cost=0.03..1.54 rows=55000 width=33)                                                                |
|                     ->  Index Scan using compress_hyper_2_25_chunk__compressed_hypertable_2_source__ts_m on compress_hyper_2_25_chunk  (cost=0.29..1.54 rows=55 width=101) |
|                           Index Cond: (source = nodetime.node)                                                                                                             |
|               ->  Custom Scan (DecompressChunk) on _hyper_1_13_chunk  (cost=0.03..1.89 rows=71000 width=33)                                                                |
|                     ->  Index Scan using compress_hyper_2_27_chunk__compressed_hypertable_2_source__ts_m on compress_hyper_2_27_chunk  (cost=0.29..1.89 rows=71 width=101) |
|                           Index Cond: (source = nodetime.node)                                                                                                             |
|               ->  Custom Scan (DecompressChunk) on _hyper_1_14_chunk  (cost=0.03..1.89 rows=71000 width=33)                                                                |
|                     ->  Index Scan using compress_hyper_2_29_chunk__compressed_hypertable_2_source__ts_m on compress_hyper_2_29_chunk  (cost=0.29..1.89 rows=71 width=101) |
|                           Index Cond: (source = nodetime.node)                                                                                                             |
|               ->  Custom Scan (DecompressChunk) on _hyper_1_16_chunk  (cost=0.03..1.89 rows=71000 width=33)                                                                |
|                     ->  Index Scan using compress_hyper_2_31_chunk__compressed_hypertable_2_source__ts_m on compress_hyper_2_31_chunk  (cost=0.29..1.89 rows=71 width=101) |
|                           Index Cond: (source = nodetime.node)                                                                                                             |
|               ->  Custom Scan (DecompressChunk) on _hyper_1_18_chunk  (cost=0.03..1.90 rows=71000 width=33)                                                                |
|                     ->  Index Scan using compress_hyper_2_33_chunk__compressed_hypertable_2_source__ts_m on compress_hyper_2_33_chunk  (cost=0.29..1.90 rows=71 width=101) |
|                           Index Cond: (source = nodetime.node)                                                                                                             |
|               ->  Index Scan using _hyper_1_20_chunk_metrics_source_idx on _hyper_1_20_chunk  (cost=0.56..459.67 rows=7667 width=33)                                       |
|                     Index Cond: ((source = nodetime.node) AND ("timestamp" > nodetime.start_time) AND ("timestamp" < nodetime.stop_time))                                  |
|               ->  Index Scan using _hyper_1_22_chunk_metrics_source_idx on _hyper_1_22_chunk  (cost=0.56..391.89 rows=6552 width=33)                                       |
|                     Index Cond: ((source = nodetime.node) AND ("timestamp" > nodetime.start_time) AND ("timestamp" < nodetime.stop_time))                                  |
|               ->  Index Scan using _hyper_1_24_chunk_metrics_source_idx on _hyper_1_24_chunk  (cost=0.56..468.73 rows=7815 width=33)                                       |
|                     Index Cond: ((source = nodetime.node) AND ("timestamp" > nodetime.start_time) AND ("timestamp" < nodetime.stop_time))                                  |
|               ->  Index Scan using _hyper_1_26_chunk_metrics_source_idx on _hyper_1_26_chunk  (cost=0.56..468.68 rows=7815 width=33)                                       |
|                     Index Cond: ((source = nodetime.node) AND ("timestamp" > nodetime.start_time) AND ("timestamp" < nodetime.stop_time))                                  |
|               ->  Index Scan using _hyper_1_28_chunk_metrics_source_idx on _hyper_1_28_chunk  (cost=0.56..468.71 rows=7815 width=33)                                       |
|                     Index Cond: ((source = nodetime.node) AND ("timestamp" > nodetime.start_time) AND ("timestamp" < nodetime.stop_time))                                  |
|               ->  Index Scan using _hyper_1_30_chunk_metrics_source_idx on _hyper_1_30_chunk  (cost=0.56..467.95 rows=7798 width=33)                                       |
|                     Index Cond: ((source = nodetime.node) AND ("timestamp" > nodetime.start_time) AND ("timestamp" < nodetime.stop_time))                                  |
|               ->  Index Scan using _hyper_1_32_chunk_metrics_source_idx on _hyper_1_32_chunk  (cost=0.56..277.03 rows=4596 width=33)                                       |
|                     Index Cond: ((source = nodetime.node) AND ("timestamp" > nodetime.start_time) AND ("timestamp" < nodetime.stop_time))                                  |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

It works if "nodetime.start_time" is replaced by a contant value:
```
> select timestamp, start_time, stop_time, node, source, value from nodetime inner join metrics on nodetime.node = metrics.source and metrics.timestamp > '2019-11-26 08:07:07+00'::timestamp with time zone and metrics.timestamp < nodetime.stop_time limit 5;

+-------------------------------+------------------------+------------------------+--------+----------+---------+
| timestamp                     | start_time             | stop_time              | node   | source   | value   |
|-------------------------------+------------------------+------------------------+--------+----------+---------|
| 2019-11-27 03:59:58.377559+00 | 2019-11-28 08:07:07+00 | 2019-11-28 08:49:24+00 | nova-1 | nova-1   | 0.0     |
| 2019-11-27 03:59:53.423055+00 | 2019-11-28 08:07:07+00 | 2019-11-28 08:49:24+00 | nova-1 | nova-1   | 0.0     |
| 2019-11-27 03:59:48.386875+00 | 2019-11-28 08:07:07+00 | 2019-11-28 08:49:24+00 | nova-1 | nova-1   | 0.0     |
| 2019-11-27 03:59:43.342377+00 | 2019-11-28 08:07:07+00 | 2019-11-28 08:49:24+00 | nova-1 | nova-1   | 0.0     |
| 2019-11-27 03:59:38.340095+00 | 2019-11-28 08:07:07+00 | 2019-11-28 08:49:24+00 | nova-1 | nova-1   | 0.0     |
+-------------------------------+------------------------+------------------------+--------+----------+---------+

> explain select timestamp, start_time, stop_time, node, source, value from nodetime inner join metrics on nodetime.node = metrics.source and metrics.timestamp > '2019-11-28 08:07:07+00'::timestamp with time zone and metrics.timestamp < nodetime.stop_time limit 5;
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

| QUERY PLAN                                                                                                                                                               |
|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Limit  (cost=0.56..0.83 rows=5 width=81)                                                                                                                                 |
|   ->  Nested Loop  (cost=0.56..2066528.86 rows=38691864 width=81)                                                                                                        |
|         ->  Seq Scan on nodetime  (cost=0.00..20.20 rows=1020 width=48)                                                                                                  |
|         ->  Append  (cost=0.56..1646.66 rows=37933 width=33)                                                                                                             |
|               ->  Index Scan using _hyper_1_32_chunk_metrics_source_idx on _hyper_1_32_chunk  (cost=0.56..582.42 rows=15228 width=33)                                    |
|                     Index Cond: ((source = nodetime.node) AND ("timestamp" < nodetime.stop_time) AND ("timestamp" > '2019-11-28 08:07:07+00'::timestamp with time zone)) |
|               ->  Index Scan using _hyper_1_30_chunk_metrics_source_idx on _hyper_1_30_chunk  (cost=0.56..874.58 rows=22705 width=33)                                    |
|                     Index Cond: ((source = nodetime.node) AND ("timestamp" < nodetime.stop_time) AND ("timestamp" > '2019-11-28 08:07:07+00'::timestamp with time zone)) |
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

It also works (but very slowly) if all index scans are disabled:
```
> set enable_indexscan = off;
> set enable_bitmapscan = off;
> select timestamp, start_time, stop_time, node, source, value from nodetime inner join metrics on nodetime.node = metrics.source and metrics.timestamp > nodetime.start_time and metrics.timestamp < nodetime.stop
                            _time limit 5;
+-------------------------------+------------------------+------------------------+--------+----------+---------+
| timestamp                     | start_time             | stop_time              | node   | source   | value   |
|-------------------------------+------------------------+------------------------+--------+----------+---------|
| 2019-11-28 08:07:42.470112+00 | 2019-11-28 08:07:07+00 | 2019-11-28 08:49:24+00 | nova-1 | nova-1   | 0.0     |
| 2019-11-28 08:07:42.470112+00 | 2019-11-28 08:07:07+00 | 2019-11-28 08:49:24+00 | nova-1 | nova-1   | 0.0     |
| 2019-11-28 08:07:47.419075+00 | 2019-11-28 08:07:07+00 | 2019-11-28 08:49:24+00 | nova-1 | nova-1   | 0.0     |
| 2019-11-28 08:07:47.419075+00 | 2019-11-28 08:07:07+00 | 2019-11-28 08:49:24+00 | nova-1 | nova-1   | 0.0     |
| 2019-11-28 08:07:57.438661+00 | 2019-11-28 08:07:07+00 | 2019-11-28 08:49:24+00 | nova-1 | nova-1   | 0.0     |
+-------------------------------+------------------------+------------------------+--------+----------+---------+
```

### PR DESCRIPTION
**Pull request by gayyappan**
Fixes #1558

### CODE REVIEW

File: tsl/src/nodes/decompress_chunk/decompress_chunk.c [hunk 1]
```diff
@@ -357,7 +357,7 @@ ts_decompress_chunk_generate_paths(PlannerInfo *root, RelOptInfo *chunk_rel, Hyp
 	compressed_rel = info->compressed_rel;
 
 	compressed_rel->consider_parallel = chunk_rel->consider_parallel;
-
+	// translate chunk_rel->baserestrictinfo
```
Comment (cevian) [inline tsl/src/nodes/decompress_chunk/decompress_chunk.c]:
```suggestion
	/* translate chunk_rel->baserestrictinfo */
```

File: tsl/src/nodes/decompress_chunk/decompress_chunk.c [hunk 2]
```diff
@@ -544,6 +544,132 @@ compressed_rel_setup_reltarget(RelOptInfo *compressed_rel, CompressionInfo *info
...
+	foreach (lc, chunk_rel->joininfo)
+	{
+		RestrictInfo *compress_ri;
+		RestrictInfo *ri = (RestrictInfo *) lfirst(lc);
+		// RestrictInfo *compress_ri = expression_tree_mutator((Node *) ri, chunk_joininfo_mutator,
+		// (void*)info);
+		Node *result = chunk_joininfo_mutator((Node *) ri, info);
+		Assert(IsA(result, RestrictInfo));
+		compress_ri = (RestrictInfo *) result;
+		// Assert(compress_ri->is_pushed_down == false); fails for exp.txt
```
Comment (cevian) [inline tsl/src/nodes/decompress_chunk/decompress_chunk.c]: why is this comment about?

File: tsl/src/nodes/decompress_chunk/planner.c [hunk 3]
```diff
@@ -313,13 +341,33 @@ decompress_chunk_plan_create(PlannerInfo *root, RelOptInfo *rel, CustomPath *pat
...
+		List *indexqual = NIL;
+		Plan *indexplan;
 		foreach (lc, clauses)
 		{
 			RestrictInfo *rinfo = lfirst_node(RestrictInfo, lc);
 			if (is_redundant_derived_clause(rinfo, ipath->indexclauses))
 				continue; /* dup or derived from same EquivalenceClass */
 			cscan->scan.plan.qual = lappend(cscan->scan.plan.qual, rinfo->clause);
 		}
+		/* the index plan below this node was already created with the
```
Comment (cevian) [inline tsl/src/nodes/decompress_chunk/planner.c]: suggested comment: joininfo clauses have to contain clauses on both compressed and decompressed attnos, but the index scans can't handle compressed attributes, so remove them from the indexscans here.

File: tsl/src/nodes/decompress_chunk/decompress_chunk.c [hunk 4]
```diff
@@ -544,6 +544,131 @@ compressed_rel_setup_reltarget(RelOptInfo *compressed_rel, CompressionInfo *info
...
+compressed_rel_setup_joininfo(RelOptInfo *compressed_rel, CompressionInfo *info)
+{
+	RelOptInfo *chunk_rel = info->chunk_rel;
+	ListCell *lc;
+	List *compress_joininfo = NIL;
+	foreach (lc, chunk_rel->joininfo)
+	{
+		RestrictInfo *compress_ri;
+		RestrictInfo *ri = (RestrictInfo *) lfirst(lc);
+		// RestrictInfo *compress_ri = expression_tree_mutator((Node *) ri, chunk_joininfo_mutator,
```
[the remark this answers is not in the document]
Comment (gayyappan) [reply tsl/src/nodes/decompress_chunk/decompress_chunk.c]: removed

File: tsl/src/nodes/decompress_chunk/decompress_chunk.c [hunk 5]
```diff
@@ -780,12 +905,21 @@ decompress_chunk_add_plannerinfo(PlannerInfo *root, CompressionInfo *info, Chunk
...
+	 * get_baserel_parampathinfo <- create_index_paths)
+	 */
+	Assert(chunk_rel->top_parent_relids != NULL);
+	compressed_rel->top_parent_relids = bms_copy(chunk_rel->top_parent_relids);
 	root->simple_rel_array[compressed_index] = compressed_rel;
 	info->compressed_rel = compressed_rel;
 
 	compressed_rel_setup_reltarget(compressed_rel, info, needs_sequence_num);
 	compressed_rel_setup_equivalence_classes(root, info);
+	// translate chunk_rel->joininfo for compressed_rel
```
[the remark this answers is not in the document]
Comment (gayyappan) [reply tsl/src/nodes/decompress_chunk/decompress_chunk.c]: changed

File: tsl/src/nodes/decompress_chunk/decompress_chunk.c (hunk 2 above)
Comment (gayyappan) [reply tsl/src/nodes/decompress_chunk/decompress_chunk.c]: this is a comment from when I didn't understand what rels have is_pushed_down set. removed assert and comment.

File: tsl/src/nodes/decompress_chunk/planner.c [hunk 6]
```diff
@@ -286,6 +286,34 @@ replace_compressed_vars(Node *node, CompressionInfo *info)
...
+static bool
+check_clause_for_compressed_attnos(Node *node, void *context)
+{
+	if (node == NULL)
+		return true;
+	if (IsA(node, Var))
+	{
+		CompressedAttnoContext *cxt = (CompressedAttnoContext *) context;
+		Var *var = (Var *) node;
+		/* constify tableoid in quals */
```
[the remark this answers is not in the document]
Comment (gayyappan) [reply tsl/src/nodes/decompress_chunk/planner.c]: removed.

File: tsl/src/nodes/decompress_chunk/decompress_chunk.c [hunk 7]
```diff
@@ -544,6 +544,129 @@ compressed_rel_setup_reltarget(RelOptInfo *compressed_rel, CompressionInfo *info
...
+			get_column_compressioninfo(context->hypertable_compression_info, column_name);
+
+		compressed_attno =
+			get_attnum(context->compressed_rte->relid, compressioninfo->attname.data);
+		compress_var->varno = context->compressed_rel->relid;
+		compress_var->varattno = compressed_attno;
+
+		return (Node *) compress_var;
+	}
+	if (IsA(node, RestrictInfo))
```
Comment (cevian) [inline tsl/src/nodes/decompress_chunk/decompress_chunk.c]:
```suggestion
	else if (IsA(node, RestrictInfo))
```

**Pull request merged**